v3/src/components/graph/models/graph-controller.ts

```diff
@@ -23,51 +31,105 @@ export interface IGraphControllerProps {
-    this.dataConfig = DataConfigurationModel.create()
     this.instanceId = props.instanceId
     this.enableAnimation = props.enableAnimation
     this.dotsRef = props.dotsRef
+    this.v2Document = props.v2Document
     if (this.dataset) {
-      this.dataConfig.setDataset(this.dataset)
+      this.graphModel.config.setDataset(this.dataset)
     }
     // Presumably a new dataset is now being used. So we have to set things up for an empty graph
     this.initializeGraph()
   }
 
   initializeGraph() {
-    const {graphModel, dataConfig, layout, dotsRef, enableAnimation, instanceId} = this
-    graphModel.setGraphProperties({
-      axes: {bottom: EmptyAxisModel.create({place: 'bottom'}), left: EmptyAxisModel.create({place: 'left'})},
-      plotType: 'casePlot', config: this.dataConfig
-    })
+    const {graphModel, layout, dotsRef, enableAnimation, instanceId, v2Document} = this,
+      dataConfig = graphModel.config
+    if (v2Document) {
+      this.processV2Document()
+    } else {
+      graphModel.setGraphProperties({
+        axes: {bottom: EmptyAxisModel.create({place: 'bottom'}),
+          left: EmptyAxisModel.create({place: 'left'})}, plotType: 'casePlot'
+      })
+      layout.setAxisScale('bottom', scaleOrdinal())
+      layout.setAxisScale('left', scaleOrdinal())
+    }
     if (dotsRef.current) {
       matchCirclesToData({
         caseIDs: dataConfig.cases, dotsElement: dotsRef.current,
         pointRadius: graphModel.getPointRadius(), enableAnimation, instanceId
       })
     }
-    layout.setAxisScale('bottom', scaleOrdinal())
-    layout.setAxisScale('left', scaleOrdinal())
+  }
+
+  processV2Document() {
+    const {graphModel, layout, /*dotsRef, enableAnimation,*/ dataset, v2Document} = this,
+      dataConfig = graphModel.config,
+      firstV2GraphComponent = v2Document?.components.find(aComp => aComp.type === 'DG.GraphView'),
+      storage = firstV2GraphComponent?.componentStorage as ICodapV2GraphStorage,
+      links = storage?._links_ || {},
+      attrTypes: Record<string, string> = {x: 'empty', y: 'empty', legend: 'empty'}
+    Object.keys(links).forEach((aKey: keyof typeof links) => {
+      if (['xAttr', 'yAttr', 'legendAttr'].includes(aKey)) {
+        const match = aKey.match(/[a-z]+/),
+          attrPlace = (match?.[0] ?? 'x') as GraphAttrPlace,
+          attrV2ID = (links[aKey] as IGuidLink<"DG.Attribute">).id,
+          attrName = v2Document?.getAttribute(attrV2ID)?.object.name,
+          attribute = dataset?.attrFromName(attrName),
+          attrID = attribute?.id ?? ''
+        graphModel.setAttributeID(attrPlace, attrID)
+        if (['x', 'y'].includes(attrPlace)) {
+          attrTypes[attrPlace] = attribute?.type ?? 'empty'
+        }
+      }
+    })
+    graphModel.setPlotType(plotChoices[attrTypes.x][attrTypes.y]);
+    ['x', 'y'].forEach((attrPlace: GraphAttrPlace) => {
+      const axisPlace = attrPlaceToAxisPlace[attrPlace as GraphAttrPlace] as AxisPlace,
```

Comment:
```suggestion
      const axisPlace = attrPlaceToAxisPlace[attrPlace] as AxisPlace,
```
This cast can go as well.